src/session-lock.c

```diff
@@ -266,25 +266,30 @@ handle_lock_unlock(struct wl_listener *listener, void *data)
 		wl_container_of(listener, manager, lock_unlock);
 	session_lock_destroy(manager);
 	manager->locked = false;
-	desktop_focus_topmost_view(manager->server);
+
+	if (manager->last_active_view) {
+		desktop_focus_view(manager->last_active_view, /* raise */ false);
+	}
```

Comment:
Should we have an else branch that focuses the topmost view here?

Reply:
Done.

I tried KDE Plasma (which doesn't use ext-session-lock) just now and I found it focuses on the last-focused window on unlock even when no window is focused on lock, so it will make a similar feeling.